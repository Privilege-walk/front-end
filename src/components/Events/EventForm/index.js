import React, { useState } from 'react';
import { connect } from 'react-redux';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import InputLabel from '@mui/material/InputLabel';
import TableHead from '@mui/material/TableHead';
import FormHelperText from '@mui/material/FormHelperText';

import Box from '@mui/material/Box';
import FormControl from '@mui/material/FormControl';
import OutlinedInput from '@mui/material/OutlinedInput';


import { createEvent } from '../../../Store/actions';


function EventForm({fetchAllEvents, createEvent}){
    const [eventName, setEventName] = useState('');
    const [errMsg, setErrMsg] = useState('');
    const [xLabelMin, setXLabelMin] = useState('');
    const [xLabelMax, setXLabelMax] = useState('');

    const handleSubmit = async (event) => {
        const action = await createEvent({ 
            newEventName: eventName,
            xLabelMin, xLabelMax
        });
        if (action.payload.status == "created"){
            fetchAllEvents();
            setEventName("");
            setErrMsg("");
            setXLabelMin("");
            setXLabelMax("");
        }else{
            setErrMsg("Unable to create the event!");
        }
    }
    
    return (<Paper 
        className="mb-3 align-items-center" 
        variant="outlined" 
        sx={{ p: '2px 4px',  display: 'flex',  flexGrow: 1, alignItems: 'center'}}
    >
        <FormControl error={errMsg.length > 0} fullWidth sx={{ m: 1 }}>
            <InputLabel htmlFor="new-event-name">New Event Name</InputLabel>
            <OutlinedInput
                id="new-event-name"
                label="New Event Name" 
                variant="filled"
                autoComplete="off"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
            />
            <FormHelperText data-testid="event-error-text">{errMsg}</FormHelperText>
        </FormControl>

        <FormControl fullWidth sx={{ m: 1 }}>
            <InputLabel htmlFor="new-event-name">X Label Min</InputLabel>
            <OutlinedInput
                id="new-event-name"
                label="New Event Name" 
                variant="filled"
                autoComplete="off"
                value={xLabelMin}
                onChange={(e) => setXLabelMin(e.target.value)}
            />
        </FormControl>

        <FormControl fullWidth sx={{ m: 1 }}>
            <InputLabel htmlFor="new-event-name">X Label Max</InputLabel>
            <OutlinedInput
                id="new-event-name"
                label="New Event Name" 
                variant="filled"
                autoComplete="off"
                value={xLabelMax}
                onChange={(e) => setXLabelMax(e.target.value)}
            />
        </FormControl>
        <Box>
            <Button 
                variant="contained"
                type="submit"
                className="ml-2"
                onClick={handleSubmit}
            >
                Create Event
            </Button>  
        </Box>
    </Paper>);
}

const mapStateToProps = state => ({});

const mapDispatchToProps = {
    createEvent
};

var EventFormContainer = connect(mapStateToProps, mapDispatchToProps)(EventForm);
export default EventFormContainer;