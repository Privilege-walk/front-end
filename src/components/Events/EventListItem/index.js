import React, { useState } from "react";
import { 
    Badge
 } from 'reactstrap';
 import { useNavigate } from "react-router-dom";
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
 import CardActions from '@mui/material/CardActions';


export default function EventListItem({ id, name, status }) {
    const navigate = useNavigate();

    function redirectToQuestions() {
        navigate('/questions', { state: {id, name, status} });
    }

    return (
        <Card className="">
            <CardContent>
            <Typography>{name} <Badge color="primary">{status}</Badge></Typography>
            
            </CardContent>
            <CardActions>
                <Button 
                    variant="outlined" 
                    size="md"
                    onClick={redirectToQuestions}
                >
                    Edit
                </Button>
            </CardActions>
            
        </Card>
    );
}