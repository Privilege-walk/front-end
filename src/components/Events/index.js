import React from "react";
import { Row, Container} from 'reactstrap';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';
import TablePagination from '@mui/material/TablePagination';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import InputLabel from '@mui/material/InputLabel';
import TableHead from '@mui/material/TableHead';
import FormHelperText from '@mui/material/FormHelperText';

import Box from '@mui/material/Box';
import FormControl from '@mui/material/FormControl';
import OutlinedInput from '@mui/material/OutlinedInput';

import { connect } from 'react-redux';
import "../App.css";
import EventListItem from "./EventListItem";
import { fetchEvents, createEvent } from '../../Store/actions';


class Events extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            newEventName: "",
            eventsList: [],
            page: 0,
            rowsPerPage: 5,
            errMsg: ""
        }
    }

    changeNewEventName = (newEventName) => {
        this.setState({newEventName});
    }

    changeEventsList = (eventsList) => {
        this.setState({eventsList});
    }

    setErrMsg = (errMsg) => {
        this.setState({errMsg});
    }

    componentDidMount() {
        this.fetchAllEvents();
    }

    handleChangePage(event, newPage){
        this.setState({page: newPage});
    };
    
    fetchAllEvents = async () => {
        const action = await this.props.fetchEvents();
        if (action.payload.error){
            this.setErrMsg("Unable to display the events list!");
        }else{
            this.changeEventsList(action.payload.events);
        }
    }

    handleSubmit = async (event) => {
        const action = await this.props.createEvent({ newEventName: this.state.newEventName});
        if (action.payload.status == "created"){
            this.fetchAllEvents();
            this.setState({newEventName: ""});
        }else{
            this.setErrMsg("Unable to create the event!");
        }
    }

    render() {
        return (
            <Container id="eventsPageId" className="mx-2">
                <Row>
                <h1 className="event">Events</h1>
            
                <Paper 
                    className="mb-3 align-items-center" 
                    variant="outlined" 
                    sx={{ p: '2px 4px',  display: 'flex',  flexGrow: 1, alignItems: 'center'}}
                >
                    <FormControl error={this.state.errMsg.length > 0} fullWidth sx={{ m: 1 }}>
                        <InputLabel htmlFor="new-event-name">New Event Name</InputLabel>
                        <OutlinedInput
                            id="new-event-name"
                            label="New Event Name" 
                            variant="filled"
                            autoComplete="off"
                            value={this.state.newEventName}
                            onChange={(e) => this.changeNewEventName(e.target.value)}
                        />
                        <FormHelperText data-testid="event-error-text">{this.state.errMsg}</FormHelperText>
                    </FormControl>
                    <Box>
                        <Button 
                            variant="contained"
                            type="submit"
                            className="ml-2"
                            onClick={this.handleSubmit}
                        >
                            Create Event
                        </Button>  
                    </Box>
                </Paper>

                </Row>
                <TableContainer component={Paper}>
                    <Table sx={{ minWidth: 650 }} aria-label="simple table">
                    <TableHead>
                        <TableRow>
                            <TableCell>
                                <h4>All events</h4>
                            </TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                    {
                        this.state.eventsList.length === 0 ?
                        (<TableRow style={{color:'gray',marginLeft:10}}>
                            <TableCell><i>No events to show.</i></TableCell>
                        </TableRow>)
                        :
                        (
                            this.state.eventsList.slice(
                                this.state.page * this.state.rowsPerPage, this.state.page * this.state.rowsPerPage + this.state.rowsPerPage
                            ).map( (item, _) => (
                                <TableRow key={item.id}>
                                    <TableCell>
                                    <EventListItem key={item.id} name={item.name} status={item.status} />
                                    </TableCell>
                                </TableRow>)
                        ))
                    }
                    </TableBody>
                    </Table>
                </TableContainer>
                <TablePagination
                    rowsPerPageOptions={[5]}
                    component="div"
                    count={this.state.eventsList.length}
                    rowsPerPage={this.state.rowsPerPage}
                    page={this.state.page}
                    onPageChange={this.handleChangePage.bind(this)}
                    />
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        token: state.authToken
    };
};

const mapDispatchToProps = {
    fetchEvents,
    createEvent
  };

var EventsContainer = connect(mapStateToProps, mapDispatchToProps)(Events);
export default EventsContainer;