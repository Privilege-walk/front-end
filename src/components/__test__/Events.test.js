import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Events from '../events';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

// const mockLoginUser = jest.fn();
jest.mock('../../Store/actions', () => ({
    loginUser: () => mockLoginUser
}));

const setup = () => {
    let utils;
    const store = mockStore({
        token: "token",
    });
    act (() => {
        utils = render(
        <Provider store={store}>
            <Events />
        </Provider>
        );
    }) 
    const newEventInput = screen.getByLabelText('New Event Name');
    const submitBtn = screen.getByText('Create Event', {selector: 'button'});
    return {  
        newEventInput,
        submitBtn,
        ...utils
    };
}
test ('renders without crashing', () => {
    let utils = setup();
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})