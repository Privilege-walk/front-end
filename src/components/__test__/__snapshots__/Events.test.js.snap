// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch events list of items 1`] = `
<DocumentFragment>
  <div
    class="mx-2 container"
    id="eventsPageId"
  >
    <div
      class="row"
    >
      <h1
        class="event"
      >
        Events
      </h1>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded mb-3 align-items-center css-lzote6-MuiPaper-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            New Event Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            data-testid="event-error-text"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            X Label Min
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            X Label Max
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root ml-2 css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Create Event
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root css-quj9j6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <h4>
                All events
              </h4>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            style="color: gray; margin-left: 10px;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <i>
                No events to show.
              </i>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          0â€“0 of 0
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without crashing: empty list 1`] = `
<DocumentFragment>
  <div
    class="mx-2 container"
    id="eventsPageId"
  >
    <div
      class="row"
    >
      <h1
        class="event"
      >
        Events
      </h1>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded mb-3 align-items-center css-lzote6-MuiPaper-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            New Event Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            data-testid="event-error-text"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            X Label Min
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="new-event-name"
          >
            X Label Max
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            variant="filled"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-event-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  New Event Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root ml-2 css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Create Event
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root css-quj9j6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <h4>
                All events
              </h4>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            style="color: gray; margin-left: 10px;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <i>
                No events to show.
              </i>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          0â€“0 of 0
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
